### ISSUE TITLE
Low-Conf Remasking: Softmax with Temperature

### ISSUE THREAD

**Issue by automenta** [ML-GSAI/LLaDA#37]
https://github.com/ML-GSAI/LLaDA/blob/57a3a4f96eccf7736415e528e481d32a4f94f7a8/generate.py#L86

# Use temperature?                                                                                                                

**Yes, it likely should for consistency and to accurately reflect the model's confidence at the given temperature.**
 - 1 Consistency with Sampling:  If you are using temperature to influence the sampling of x0 in logitSample, then the "confidence" calculation should also be reflective of that temperature.  If you sample with a high temperature (making the choices more random), the         
   confidence score should also reflect this increased uncertainty.                                                                               
 - 2 Meaning of "Confidence":  Without temperature, the confidence score is based on the raw logits. However, the actual token        
   selection (x0) is influenced by temperature through add_gumbel_noise.  Therefore, the confidence score calculated without temperature might not
   accurately represent the model's certainty in the token after temperature scaling has been applied for sampling.                               
 - 3 Intuitive Behavior:  If you increase the temperature, you expect the model to be less "confident" in its choices.  If it doesn't use   
   temperature, the confidence scores might not decrease as much as they should when temperature increases, potentially leading to less effective low_confidence remasking at higher temperatures.                                                                                               

```
     p = F.softmax(logits / temperature if temperature > 0 else logits, dim=-1)
     x0_p = gather(p, dim=-1, index=x0.unsqueeze(-1)).squeeze(-1)
```

**Comment by NieShenRuc** [comment]
Thank you for your attention! 

We think your explanation of temperature is reasonable. However, how to set it optimally for downstream tasks still requires experimental observation.

**Comment by automenta** [comment]
everyone knows its value now
